File: competitive_programming/codeforces/06_javascript/339a_helpful_maths.js
function helpfulMaths(expression) {
  return expression.split("+").sort().join("+");
}

if (require.main === module) {
  if (helpfulMaths("3+2+1") !== "1+2+3") throw new Error("helpful maths failed");
  if (helpfulMaths("1+1+3+1+3") !== "1+1+1+3+3") throw new Error("helpful maths failed");
  console.log("339A helpful maths ok");
}

File: competitive_programming/codeforces/09_rust/339a_helpful_maths.rs
fn helpful_maths(expression: &str) -> String {
    let mut parts: Vec<&str> = expression.split('+').collect();
    parts.sort();
    parts.join("+")
}

fn main() {
    assert_eq!(helpful_maths("3+2+1"), "1+2+3");
    assert_eq!(helpful_maths("1+1+3+1+3"), "1+1+1+3+3");
    println!("339A helpful maths ok");
}
